monai/handlers/tensorboard_handler.py

```diff
@@ -0,0 +1,141 @@
+        Write epoch level events, default values are from ignite state.metrics dict.
+
+        Args:
+            engine (ignite.engine): Ignite Engine, it can be a trainer, validator or evaluator.
+
+        """
+        if self.epoch_event_writer is not None:
+            self.epoch_event_writer(engine, self._writer)
+        else:
+            self._default_epoch_writer(engine, self._writer)
+
+    def iteration_completed(self, engine: Engine):
+        """handler for train or validation/evaluation iteration completed Event.
+        Write iteration level events, default values are from ignite state.logs dict.
+
+        Args:
+            engine (ignite.engine): Ignite Engine, it can be a trainer, validator or evaluator.
+
+        """
+        if self.iteration_event_writer is not None:
+            self.iteration_event_writer(engine, self._writer)
+        else:
+            self._default_iteration_writer(engine, self._writer)
+
+    def _default_epoch_writer(self, engine: Engine, writer: SummaryWriter):
+        """Execute epoch level event write operation based on Ignite engine.state data.
+        Default is to write the values from ignite state.metrics dict.
+
+        Args:
+            engine (ignite.engine): Ignite Engine, it can be a trainer, validator or evaluator.
+            writer (SummaryWriter): TensorBoard writer, created in TensorBoardHandler.
+
+        """
+        current_epoch = self.global_epoch_transform() if self.global_epoch_transform is not None \
+            else engine.state.epoch
+        summary_dict = engine.state.metrics
+        for name, value in summary_dict.items():
+            writer.add_scalar(name, value, current_epoch)
+
+        show_labels = self.batch_transform(engine.state.batch)[1]
+        show_outputs = self.output_transform(engine.state.output)[0].detach().cpu()
+        if show_labels is not None and show_outputs is not None:
+            ones = torch.ones(show_labels[0].shape, dtype=torch.int32)
+            if len(show_labels.shape) == 5:
+                assert show_labels.shape[1] == show_outputs.shape[1] == 1, \
+                    '3D images must select only 1 channel to show.'
+                for i in range(len(show_labels)):
+                    img2tensorboard.add_animated_gif(writer, 'output' + str(i), show_outputs[i], 64, 255, current_epoch)
+                    label_tensor = torch.where(show_labels[i] > 0, ones, show_labels[i])
+                    img2tensorboard.add_animated_gif(writer, 'label' + str(i), label_tensor, 64, 255, current_epoch)
+            elif len(show_labels.shape) == 4:
+                for i in range(len(show_labels)):
+                    writer.add_image('output' + str(i), show_outputs[i], current_epoch)
+                    label_tensor = torch.where(show_labels[i] > 0, ones, show_labels[i])
+                    writer.add_image('label' + str(i), label_tensor, current_epoch)
```

Comment:
should we raise an error when the shape is not 4 or 5?

Reply:
fixed in the latest patch.
Thanks.